thread_setup_unix.c

```diff
@@ -9,6 +9,9 @@ static pthread_mutex_t *mutex_buf = NULL;
 
 static pthread_key_t destructor_key;
 
+/* Used in pthread_setspecific. See https://github.com/microsoft/go/issues/1305. */
+static char stub;
```

Comment:
Does this need to exist at the global scope, or could `static char stub;` or `char stub;` be declared just before the call?

If there's no difference, I think putting the declaration inside `thread_id` would be a bit clearer. And IMO putting it on the stack (if possible) raises the fewest questions from a reader's perspective. 😄

Reply:
Looks putting the variable in the function scope also works, and it does look clearer. I'll keep the `static` keyword, though, I would rather not pass a pointer to a stack-allocated variable that can be used outside the function frame, just in case it is used somewhere 😅.